linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/IstioIdentifier.scala

```diff
@@ -23,14 +24,36 @@ class IstioIdentifier(pfx: Path, baseDtab: () => Dtab, routeCache: RouteCache, c
     }
   }
 
+  def compareStringMatch(headerValue: String, stringMatch: StringMatch): Boolean = {
+    stringMatch.`matchType` match {
+      case Some(OneofMatchType.Exact(value)) => headerValue == value
+      case Some(OneofMatchType.Prefix(pfx)) => headerValue.startsWith(pfx)
+      case Some(OneofMatchType.Regex(r)) => headerValue.matches(r)
+      case None => throw new IllegalArgumentException("stringMatch missing matchType")
+    }
+  }
+
+  def compareMatchConditions(req: Request, matchCondition: MatchCondition): Boolean = {
+    val matchesHeaders = matchCondition.`httpHeaders`.map {
+      case (headerName, stringMatch) =>
+        println(headerName, stringMatch)
+        // return false if the headerName does not appear in the request's headerMap
+        req.headerMap.get(headerName).map(compareStringMatch(_, stringMatch)).getOrElse(false)
+    }.fold(true)(_ && _)
+    //TODO: add other match conditions
+    matchesHeaders
+  }
+
   override def apply(req: Request): Future[RequestIdentification[Request]] = {
     req.host match {
       case Some(host) =>
         Future.join(clusterCache.get(host), routeCache.getRules).map {
           case (Some(Cluster(dest, port)), rules: Map[String, RouteRule]) =>
             val filteredRules: Seq[(String, RouteRule)] = rules.filter {
-              //TODO: add more route conditions
-              case (_, r) => r.`destination` == Some(dest)
+              case (_, r) if r.`destination` == Some(dest) =>
+                // return true if no match conditions were defined on the route-rule
+                r.`match`.map(compareMatchConditions(req, _)).getOrElse(true)
```

Comment:
could do match instead of .map.getOrElse here as well if you think it improves clarity.